test/natvis_tests.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright 2024 Braden Ganetsky
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+
+// This test applies to MSVC only. This is a file for manual testing.
+// Run this test and break manually at the variable called `break_here`.
+// Inspect the variables using the Visual Studio debugger to test correctness.
+
+#if 0 // Change to `#if 1` to test turning off SIMD optimizations
+#define BOOST_UNORDERED_DISABLE_SSE2
+#define BOOST_UNORDERED_DISABLE_NEON
+#endif
+
+#define BOOST_UNORDERED_ENABLE_STATS
+
+#include <boost/core/lightweight_test.hpp>
+
+#if defined(BOOST_MSVC)
+
+#include <boost/interprocess/allocators/allocator.hpp>
+#include <boost/interprocess/managed_shared_memory.hpp>
+#include <boost/unordered/concurrent_flat_map.hpp>
+#include <boost/unordered/concurrent_flat_set.hpp>
+#include <boost/unordered/unordered_flat_map.hpp>
+#include <boost/unordered/unordered_flat_set.hpp>
+#include <boost/unordered/unordered_map.hpp>
+#include <boost/unordered/unordered_node_map.hpp>
+#include <boost/unordered/unordered_node_set.hpp>
+#include <boost/unordered/unordered_set.hpp>
+#include <boost/uuid/random_generator.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <string>
+#include <typeinfo>
+
+// Prevent any "unused" errors
+template <class... Args> void use(Args&&...) {}
+
+using map_value_type = std::pair<const std::string, int>;
+using set_value_type = std::string;
+
+template <class Tester> void natvis_test(Tester& tester)
+{
+  // clang-format off
+    auto fca_map      = tester.template construct_map<boost::unordered_map>();
+    auto fca_multimap = tester.template construct_map<boost::unordered_multimap>();
```

Comment:
Idiomatically, the test suite would pass `T* p = nullptr;` to these kinds of functions so the type could be deduced from the argument.

Not sure if this would wind up looking better or not, but maybe worth considering given how the test suite is mostly structured now.